test/helpers/powervs.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/authenticator"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/utils"
+)
+
+var (
+	pollingInterval                = time.Second * 30
+	powerVSInstanceDeletionTimeout = time.Minute * 10
+)
+
+const (
+	powervsInstanceDoesNotExist  = "pvm-instance does not exist"
+	powervsInstanceNotFound      = "could not be found"
+	powervsInstanceStateDeleting = "deleting"
+)
+
+// CheckPowerVSInstances verifies if the Powervs instances are deleted
+// before proceeding with the next cluster creation.
+func CheckPowerVSInstances(serviceInstanceID string) error {
```

Comment:
Just a thought, it can be this anyway, If possible its better to rename this function because i think
1. PowerVSInstances can be PowerVS server instance or service instance, Here we are looking for server instances(VM).
2. We are only checking for deleting state, may be some name related to delete would directly indicate the purpose of function.

Reply:
yes, can be renamed, PTAL.